### PR TITLE
Move this to the main dev branch

### PR DESCRIPTION
**Pull request by cloudkucooland**
I've had too many problems rebasing this, so I'm going to work on the feature branch. Since this touches so many other things, it'll be better to have it more visible.

### CODE REVIEW

File: src/code/team.js [hunk 1]
```diff
@@ -8,40 +8,62 @@ export default class WasabeeTeam {
...
+        console.error(e);
+        return;
       }
     }
 
-    this.agents = new Array();
     this.id = data.id;
     this.name = data.name;
-    this.fetched = Date.now();
+    this.feteched = Date.now();
```
Comment (cloudkucooland) [inline src/code/team.js:18]: surely this is a typo?

File: src/code/wd.js [hunk 2]
```diff
@@ -1,9 +1,8 @@
 import WasabeeMe from "./me";
```
Comment (cloudkucooland) [inline src/code/wd.js:1]: None of this should have been done on the IDB_CACHES branch, clearly I'm not smart enough to deal with maintaining multiple working branches at once....

Comment (cloudkucooland) [reply src/code/wd.js:1]: wait, yes, I did some IDB work here... ugh..

File: src/code/dialogs/onlineAgentList.js [hunk 3]
```diff
@@ -85,11 +85,40 @@ const OnlineAgentList = WDialog.extend({
...
+    let cursor = await tx.store.index("date").openCursor(range);
+    while (cursor) {
+      a.push(new WasabeeAgent(cursor.value, 0, false));
+      cursor = await cursor.continue();
     }
+
     this._table.items = a;
   },
+
+  _last_hour: function () {
```
Comment (cloudkucooland) [inline src/code/dialogs/onlineAgentList.js:99]: This looks ugly, surely js has a better way of doing this, right?

File: src/code/sortable.js [hunk 4]
```diff
@@ -74,29 +77,17 @@ export default class Sortable {
...
+        let value = field.value(obj);
+        if (typeof value.then === "function") value = await value; // resolve promises
+        data.values.push(value);
 
         // calculate sortValue using the field's rules if required
         let sortValue = value;
-        if (field.sortValue) sortValue = field.sortValue(value, obj);
-        data.sortValues.push(sortValue);
+        if (field.sortValue) {
+          sortValue = field.sortValue(value, obj);
```
Comment (le-jeu) [inline src/code/sortable.js:87]: isn't `await field.sortValue(value, obj)` enough ? (I don't know)

Comment (le-jeu) [reply src/code/sortable.js:87]: same for `field.value(obj)`

**Comment by le-jeu** [comment]
Some wasabee tables were rendered even if some data is missing, where promise resolution triggered UI updates
This may add some delay to the rendering (since all promise must be resolved) if I understand correctly.
A contrario, this can reduce the number of UI update (wasabee hook), but I didn't inspect the whole PR yet

**Comment by cloudkucooland** [comment]
Merging. Now. I will address the comments in the next set of commits.

**Pull request merged**